collect_app/src/main/java/org/odk/collect/android/tasks/InstanceGoogleSheetsUploader.java

```diff
@@ -391,10 +428,47 @@ private void createSheetsIfNeeded(TreeElement element) throws UploadException {
         return answers;
     }
 
-    private List<Object> getColumnTitles(TreeElement element) {
+    /**
+     * Strip the Altitude and Accuracy from a location String and add them as separate column if
+     * the column titles exist.
+     *
+     * @param answers - HashMap of current Answers
+     * @param elementTitle - Current header title
+     * @param answer - String that holds the Lat/Long/Altitude/Accuracy
+     */
+    private void parseAltitudeAndAccuracy(@NonNull Map<String, String> answers, @NonNull List<Object> columnTitles, @NonNull String elementTitle, @NonNull String answer) {
+
+        // Accuracy
+        int accuracyLocation = answer.lastIndexOf(' ');
+        String accuracyStr = answer.substring(accuracyLocation).trim();
+        answer = answer.substring(0, accuracyLocation).trim();
+        final String accuracyTitle = elementTitle + accuracyTitlePostfix;
+        if (columnTitles.contains(accuracyTitle)) {
+            answers.put(accuracyTitle, accuracyStr);
+        }
+
+        // Altitude
+        int altitudeLocation = answer.lastIndexOf(' ');
+        String altitudeStr = answer.substring(altitudeLocation).trim();
+        answer = answer.substring(0, altitudeLocation).trim();
+        final String altitudeTitle = elementTitle + altitudeTitlePostfix;
+        if (columnTitles.contains(altitudeTitle)) {
+            answers.put(altitudeTitle, altitudeStr);
+        }
+
+        // Put the modified geo location (Just lat/long) into the answers Map
+        answers.put(elementTitle, answer);
+    }
+
+    private List<Object> getColumnTitles(TreeElement element, boolean newSheet) {
         List<Object> columnTitles = new ArrayList<>();
         for (TreeElement child : getChildElements(element)) {
-            columnTitles.add(getElementTitle(child));
+            final String elementTitle = getElementTitle(child);
+            columnTitles.add(elementTitle);
+            if (newSheet && child.getDataType() == org.javarosa.core.model.Constants.DATATYPE_GEOPOINT) {
```

Comment:
why do we add those two columns only if it's a new sheet?

Reply:
Yes, to allow existing sheets to work in exactly the same way.